validator-api/src/nymd_client.rs

```diff
@@ -281,20 +277,27 @@ impl<C> Client<C> {
         self.execute_multiple_with_retry(msgs, Default::default(), memo)
             .await?;
 
-        // reward rest of delegators
-        let mut remaining_delegators = node.total_delegations - MIXNODE_DELEGATORS_PAGE_LIMIT;
-        let delegator_rewarding_msg = (
-            node.to_next_delegator_reward_execute_msg(interval_id),
-            vec![],
-        );
-        for _ in 0..further_calls {
-            let delegators_in_call = remaining_delegators.min(MIXNODE_DELEGATORS_PAGE_LIMIT);
-            let msgs = vec![delegator_rewarding_msg.clone()];
-            let memo = format!("rewarding another {} delegators", delegators_in_call);
-            self.execute_multiple_with_retry(msgs, Default::default(), memo)
-                .await?;
-
-            remaining_delegators -= MIXNODE_DELEGATORS_PAGE_LIMIT;
+        // reward rest of delegators (if applicable)
```

Comment:
In think we can just change that sub to saturating sub, ie:
```
let mut remaining_delegators = node.total_delegations.saturating_sub(MIXNODE_DELEGATORS_PAGE_LIMIT);
```

Reply:
I was actually about to apply your suggestion, but then realised it wouldn't totally solve the problem, since `further_calls` variable is actually calculated incorrectly (in the case when `node.total_delegations % MIXNODE_DELEGATORS_PAGE_LIMIT == 0`). However, I used your `saturating_sub` idea to simplify the loop